src/lib.rs

```diff
@@ -941,6 +943,60 @@ pub mod invariant {
         fn get_fee_tiers(&self) -> Vec<FeeTier> {
             self.fee_tiers.get_all()
         }
+
+        #[ink(message)]
+        fn get_tickmap(&self, pool_key: PoolKey, current_tick: i32) -> Vec<u64> {
+            let tick_spacing = pool_key.fee_tier.tick_spacing;
+            let mut tickmap_slice: Vec<u64> = vec![];
+
+            let (current_chunk_index, _) = tick_to_position(current_tick, tick_spacing);
+            let current_chunk = self
+                .tickmap
+                .bitmap
+                .get((current_chunk_index, pool_key))
+                .unwrap_or(0);
+            tickmap_slice.push(current_chunk);
+
+            let mut step = 1;
+            let max_chunks_queried = 2047;
+            while tickmap_slice.len() < max_chunks_queried {
+                if current_chunk_index + step > MAX_CHUNK {
+                    let chunk = self
+                        .tickmap
+                        .bitmap
+                        .get((current_chunk_index - step, pool_key))
+                        .unwrap_or(0);
+                    step += 1;
+                    tickmap_slice.insert(0, chunk);
```

Comment:
make sure about the order of these chunks. It should be the same order as in the tickmap state (lower index in tickmap means a lower index in the result vector)